### PR TITLE
docs: add enum documentation page

### PR DESCRIPTION
**Pull request by TomChv**
Fixes https://github.com/dagger/dagger/issues/8071

### CODE REVIEW

File: docs/current_docs/manuals/developer/enums.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+</TabItem>
+<TabItem value="Python">
+
+Here is an example of a `trivy` Dagger module with a function named `scan` that will take as arguments the reference of an image and a severity
+defined as an enum `Severity`:
+
+```python file=./snippets/enums/python/main.py
+```
+
+</TabItem>
```
Comment (helderco) [inline docs/current_docs/manuals/developer/enums.mdx:39]: I feel that Python needed a little more explanation (e.g., `dagger.Enum` is not actually required) that could be added in a note here, but wrote it in the [reference docs](https://dagger-io--8104.org.readthedocs.build/en/8104/module.html#dagger.enum_type) instead.

**Review by helderco — COMMENTED** [review]
Generally looks ok, just needs a few touchups in English, if @vikram-dagger can help with that.

File: docs/current_docs/manuals/developer/enums.mdx (hunk 1 above)
Comment (TomChv) [reply docs/current_docs/manuals/developer/enums.mdx:39]: I see that reference isn't doc isn't published yet, so I'll add a note and we can add the reference later.
Or maybe we can release an update for the Python reference doc before?

Comment (TomChv) [reply docs/current_docs/manuals/developer/enums.mdx]: Added note in [`a6c5321` (#8104)](https://github.com/dagger/dagger/pull/8104/commits/a6c532179099348776ba3f4cd7e383d58a88889a)

File: docs/current_docs/manuals/developer/enums.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+</Tabs>
+
+
+:::note
+Enumerations will be displayed when calling `--help` on a Dagger Function.
+:::
+
+Here's an example of displaying a Dagger Function helper on this method:
```
[the remark this answers is not in the document]
Comment (TomChv) [reply docs/current_docs/manuals/developer/enums.mdx]: Fixed in [`a6c5321` (#8104)](https://github.com/dagger/dagger/pull/8104/commits/a6c532179099348776ba3f4cd7e383d58a88889a)

File: docs/current_docs/manuals/developer/enums.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+# Enums
+
+
+Dagger supports custom enumeration types, this is useful to restrict possible values for an arguments.
+
+:::note
+Following the [GraphQL specification](https://spec.graphql.org/October2021/#Name), enums are represented as strings in the GraphQL schema and follows the following rules:
+- The value cannot start with digit and can only be composed of letter, digit or _
+- Name are case sensitive, and by convention should be in screaming case
+:::
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/enums.mdx:19]: Looking at this now @helderco @TomChv

File: docs/current_docs/manuals/developer/enumerations.mdx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Enumerations
+
+Dagger supports custom enumeration (enum) types, which can be used to restrict possible values for a string argument.
+
+:::note
+Following the [GraphQL specification](https://spec.graphql.org/October2021/#Name), enums are represented as strings in the Dagger API GraphQL schema and follow these rules:
+- Enum values cannot start with digits, and can only be composed of alphabets, digits or `_`.
```
Comment (vikram-dagger) [inline docs/current_docs/manuals/developer/enumerations.mdx:16]:
```suggestion
- Enum names cannot start with digits, and can only be composed of alphabets, digits or `_`.
```

To be checked

File: docs/current_docs/manuals/developer/enumerations.mdx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+- Enum names are case-sensitive, and by convention should be upper-cased.
```
Comment (vikram-dagger) [inline docs/current_docs/manuals/developer/enumerations.mdx:17]:
```suggestion
- Enum values are case-sensitive, and by convention should be upper-cased.
```

**Review by vikram-dagger — APPROVED** [review]

**Comment by vikram-dagger** [comment]
LGTM, see my comments regarding names/values before merging

**Comment by TomChv** [comment]
Added your comment @vikram-dagger, I'm merging that PR once CI's green

**Pull request merged**